### PR TITLE
Repair broken tests with matplotlib 3.3

### PR DESCRIPTION
**Pull request by rebeccabilbro**
Matplotlib 3.3 has a number of [API changes](https://matplotlib.org/3.3.0/api/api_changes.html) which are currently breaking over 40 of our image comparison tests, particularly for scatterplots and barcharts. I'm still digging through all the changes and deprecations on their end, but this PR updates the baseline images such that they pass on my machine (a mac) with matplotlib 3.3. Will have to see how things go in CI.

On top of the image comparison failures, there appear to be some additional issues with new/deprecated rcParams in MPL 3.3 that may require more research to get all of our tests passing again. 

🎃

### CODE REVIEW

File: tests/test_style/test_rcmod.py [hunk 1]
```diff
@@ -38,9 +38,25 @@ class RCParamTester(VisualTestCase):
     """
 
     excluded_params = {
-        "backend",  # This cannot be changed by manipulating rc
+        "backend",  # This cannot be changed by manipulating rc,
```
Comment (bbengfort) [inline tests/test_style/test_rcmod.py:41]: gah, how did that stray comma get there? is it worth it to run CI again just to get rid of it?

File: tests/test_style/test_rcmod.py [hunk 2]
```diff
@@ -38,9 +38,25 @@ class RCParamTester(VisualTestCase):
...
+        "webagg.port_retries",
+        "webagg.open_in_browser",
+        "backend_fallback",
+        "toolbar",
+        "timezone",
+        "datapath",
+        "figure.max_open_warning",
+        "savefig.directory",
+        "tk.window_focus",
+        "docstring.hardcopy",
```
Comment (bbengfort) [inline tests/test_style/test_rcmod.py:59]: @rebeccabilbro did a lot of work regenerating baseline images for binning, class balance, feature correlation, dispersion plot, freqdist, postag, and tsne - but somehow adding these lines to the exclude function managed to revert them to their original state (well fingers crossed, it worked on my MacBook, but not sure about CI). The changes seemed to be related to grid lines and bar colors. 

current working hypothesis - we were setting one of these values unintentionally and as a result, matplotlib was ignoring all of our styles, by excluding them, the styles work as expected? 

But you know … that's just a hypothesis, I really have no clue what is actually happening.

Comment (bbengfort) [reply tests/test_style/test_rcmod.py:59]: @rebeccabilbro I did build the docs and everything was fine - and the figures generated by the plot directive were as I expected, RE grid lines and colors.

**Comment by bbengfort** [comment]
AppVeyor tests appear to have timed out, restarting tests.

File: tests/test_style/test_rcmod.py (hunk 2 above)
Comment (bbengfort) [reply tests/test_style/test_rcmod.py:59]: Still flummoxed by this … 🤷

**Comment by rebeccabilbro** [comment]
Thank you so much for your help with this @bbengfort!

I'm going to try to work through and remove the unnecessary tols before merging (hopefully Appveyor miniconda will get it's act together in the meantime).

As for your question about kmeans, I think that it may actually be a sklearn issue related to [changes in their kmeans implementation that we discovered here](https://github.com/DistrictDataLabs/yellowbrick/pull/1068).

This was one of the diffs I got:
test_integrated_kmeans_elbow-failed-diff


Here's kmeans elbow with sklearn 0.22.2_post1:
elbow_sklearn_0_22_2_post1

sklearn 0.23:
elbow_sklearn_0_23_0

sklearn 0.23.1:
elbow_sklearn_0_23_1

sklearn 0.23.2 (latest version):
elbow_sklearn_0_23_2

**Comment by bbengfort** [comment]
@rebeccabilbro ok - awesome, glad that we can at least trace the changes in kelbow to scikit-learn; thanks for tracking that down!

**Comment by rebeccabilbro** [comment]
@bbengfort unfortunately I eventually had to remove the miniconda test from our Appveyor test matrix because I was encountering persistent problems where the tests would hang on the miniconda build during the `tests\test_utils\test_types.py` tests, and Appveyor would eventually time out. This happened regardless of experimentation with the Appveyor image, the Python version, and the Miniconda version. I'll open an issue to try to Miniconda at some point in the future when Appveyor has resolved the image issues on their end.

**Pull request merged**